language/move-lang/src/parser/syntax.rs

```diff
@@ -163,14 +213,18 @@ fn parse_name_begin_args<'input>(tokens: &mut Lexer<'input>) -> Result<Spanned<S
 }
 
 // Parse a name or a name with type arguments.
-// The return value is a tuple containing the name and a flag to
-// indicate whether there are type arguments to parse.
+// The return value is a tuple containing the name and an optional
+// value that indicates the source location of the "<" if the name
+// has type arguments.
 fn parse_name_maybe_args<'input>(
     tokens: &mut Lexer<'input>,
-) -> Result<(Spanned<String>, bool), Error> {
+) -> Result<(Spanned<String>, Option<usize>), Error> {
```

Comment:
It's a little weird that its a usize. Would it be possible to make it a loc of some sort? (even if its just a type alias?)

Reply:
The usize is consistent with other things in the parser, e.g., the return type of the lexer's start_loc() function. I suppose I could add a LocOffset type alias but then that would be inconsistent with Span, which uses usize directly. Related to this, I took a quick look yesterday at removing the NameBeginArgs tokens and it seems like it might not be very hard after all. My preference would be to try fixing that and then this code would just go away. WDYT?

Reply:
I rebased this and that code is gone now.